dashboard/test/controllers/admin_users_controller_test.rb

```diff
@@ -257,27 +260,121 @@ class AdminUsersControllerTest < ActionController::TestCase
   test 'delete_progress_form returns correct data' do
     sign_in @admin
 
-    user = create(:student, username: 'test_student')
-    script = create(:script, :with_levels, levels_count: 3)
-    level1 = script.script_levels.first.level
-    level2 = script.script_levels.second.level
-    level3 = script.script_levels.third.level
+    UserLevel.create!(user: @user, script: @script, level: @level1, best_result: 0)
+    UserLevel.create!(user: @user, script: @script, level: @level2, best_result: 10)
+    UserLevel.create!(user: @user, script: @script, level: @level3, best_result: 100)
 
-    UserScript.create!(user: user, script: script)
-    UserLevel.create!(user: user, script: script, level: level1, best_result: 0)
-    UserLevel.create!(user: user, script: script, level: level2, best_result: 10)
-    UserLevel.create!(user: user, script: script, level: level3, best_result: 100)
-
-    get :delete_progress_form, params: {user_id: user.id, script_id: script.id}
+    get :delete_progress_form, params: {user_id: @user.id, script_id: @script.id}
     assert_response :ok
     assert_select "strong" do  |elements|
       assert_equal 4, elements.length
-      assert_equal user.username, elements[1].text
-      assert_equal script.name, elements[2].text
+      assert_equal @user.username, elements[1].text
+      assert_equal @script.name, elements[2].text
       assert_equal 3, elements[3].text.to_i   # count of user_level rows
     end
   end
 
+  test "delete_progress returns error if not admin" do
+    sign_in @not_admin
+    post :delete_progress, params: {user_id: @user.id, script_id: @script.id}
+    assert_response :forbidden
+  end
+
+  test "delete_progress raises error if reason is empty" do
+    sign_in @admin
+    assert_raises(ActionController::ParameterMissing) do
+      post :delete_progress, params: {user_id: @user.id, script_id: @script.id, reason: ''}
+    end
+  end
+
+  test "delete_progress deletes script progress" do
+    sign_in @admin
+
+    UserScript.create!(user: @user, script: @script)
+    assert_equal 1, @user.user_scripts.count
+
+    post :delete_progress, params: {user_id: @user.id, script_id: @script.id, reason: 'Testing'}
+    @user.reload
+    assert_equal 0, @user.user_scripts.count
+  end
+
+  test "delete_progress deletes level progress" do
+    sign_in @admin
+
+    UserLevel.create!(user: @user, script: @script, level: @level1, best_result: 100)
+    UserLevel.create!(user: @user, script: @script, level: @level2, best_result: 50)
+    assert_equal 2, @user.user_levels_by_level(@script).count
+
+    post :delete_progress, params: {user_id: @user.id, script_id: @script.id, reason: 'Testing'}
+    assert_equal 0, @user.user_levels_by_level(@script).count
+  end
+
+  test "delete_progress deletes channel tokens" do
+    # TODO: Write this test after the work to add script_ids to channel tokens is completed.
```

Comment:
This could be written now since there is a script_id column on channel tokens.

Reply:
I wanted to use the same query methods that are used in the primary codepaths if possible and those haven't been updated to look at `script_id` yet, right?